scripts/competitiond.py

```diff
@@ -0,0 +1,580 @@
+            sys.exit(1)
+        return config
+
+    def ensure_log_worksheet_private(self):
+        """
+        Ensure that the leaderboard worksheet is private, so that all bundles
+        created on it are automatically private.
+        """
+        # Get public group info
+        public = self.client.fetch('groups', 'public')
+
+        # Set permissions
+        self.client.create('worksheet-permissions', {
+            'group': JsonApiRelationship('groups', public['id']),
+            'worksheet': JsonApiRelationship('worksheets', self.config['log_worksheet_uuid']),
+            'permission': 0,
+        })
+
+    def make_public_readable(self, bundle):
+        # Get public group info
+        public = self.client.fetch('groups', 'public')
+
+        # Set permissions
+        self.client.create('bundle-permissions', {
+            'group': JsonApiRelationship('groups', public['id']),
+            'bundle': JsonApiRelationship('bundles', bundle['id']),
+            'permission': 1,
+        })
+
+    def untag(self, bundles, tag):
+        self.client.update('bundles', [{
+            'id': b['id'],
+            'metadata': {'tags': [t for t in b['metadata']['tags'] if t != tag]}
+        } for b in bundles])
+
+    def collect_submissions(self):
+        logger.debug("Collecting latest submissions")
+        # Fetch all submissions
+        all_submissions = self.client.fetch('bundles', params={
+            'keywords': [
+                'tags={submission_tag}'.format(**self.config),
+                'created=.sort-',
+                '.limit=1500',  # should not exceed 1500 submissions ever
+            ]
+        })
+
+        # Drop all but the latest submission for each user.
+        submissions = {}
+        for bundle in reversed(all_submissions):
+            owner_id = bundle['owner']['id']
+            created = bundle['metadata']['created']
+            if owner_id not in submissions or \
+                    created > submissions[owner_id]['metadata']['created']:
+                submissions[owner_id] = bundle
+
+        # Fetch latest predict run bundles
+        last_tests = self.client.fetch('bundles', params={
+            'keywords': [
+                'tags={predict[tag]}'.format(**self.config),
+                '.limit={max_leaderboard_size}'.format(**self.config),
+            ]
+        })
+
+        # Build map from submitter_user_id -> UNIX timestamps of submissions, sorted
+        submission_times = defaultdict(list)
+        submission_ids = set()
+        for predict_bundle in last_tests:
+            submit_info = json.loads(predict_bundle['metadata']['description'])
+            timestamp = predict_bundle['metadata']['created']
+            submission_ids.add(submit_info['bundle_id'])
+            submission_times[submit_info['submitter_id']].append(timestamp)
+
+        # Drop submission if user has exceeded their quota
+        now = time.time()
+        one_day_ago = now - DAY_SECONDS
```

Comment:
Do we want to make the `DAY_SECOND` variable user configurable? For example, running a competition where users can submit once a week.